### PR TITLE
Multisend fix

### LINKED ISSUE #101: Update condition when creating a transaction: single vs batch
## Description
Update the condition inside `createTransaction(...)` that decides if the SDK will send a single or a batch transaction.
When passing an array of transaction as a parameter, take into account:
 - If the length of the array is 0
 - If the length of the array is 1 -> create single transaction

Check: https://gist.github.com/cmdzro/0547725ca6faa2067e7edf3c70b697f1#gistcomment-3954987

### PR DESCRIPTION
**Pull request by germartinez**
## What it solves
Resolves #101 

## How this PR fixes it
- Considers that transaction arrays with length = 1 are single transactions, not Multisend.
- Does not accept transaction arrays with length = 0
- Tests are updated to consider these use cases

### CODE REVIEW

File: packages/safe-core-sdk/src/Safe.ts [hunk 1]
```diff
@@ -260,11 +261,17 @@ class Safe {
   ): Promise<SafeTransaction>
   async createTransaction(
     safeTransactions: SafeTransactionDataPartial | MetaTransactionData[],
-    options?: SafeTransactionDataPartial
+    options?: SafeTransactionOptionalProps
   ): Promise<SafeTransaction> {
-    if (safeTransactions instanceof Array) {
+    const isTxArray = Array.isArray(safeTransactions)
+    if (isTxArray && (safeTransactions as MetaTransactionData[]).length === 0) {
```
Comment (iamacook) [inline packages/safe-core-sdk/src/Safe.ts:267]: Will this always be `MetaTransactionsData[]`?

Comment (iamacook) [reply packages/safe-core-sdk/src/Safe.ts:267]: As well as the other casts you make in this function?

Comment (iamacook) [reply packages/safe-core-sdk/src/Safe.ts:267]: You could also just cast it once under a new variable, no?

**Review by iamacook — COMMENTED** [review]
Looking good, just one small thing

File: packages/safe-core-sdk/src/Safe.ts (hunk 1 above)
Comment (germartinez) [reply packages/safe-core-sdk/src/Safe.ts:267]: It can be `MetaTransactionsData[]` or `SafeTransactionDataPartial`

Comment (iamacook) [reply packages/safe-core-sdk/src/Safe.ts:267]: Maybe make a type checking function that returns `value is X`?

**Review by iamacook — APPROVED** [review]
🚀

**Pull request merged**